ml-agents/mlagents/trainers/stats.py

```diff
@@ -199,8 +202,22 @@ def _maybe_create_summary_writer(self, category: str) -> None:
                 basedir=self.base_dir, category=category
             )
             os.makedirs(filewriter_dir, exist_ok=True)
+            if self._clear_past_data:
+                self._delete_all_events_files(filewriter_dir)
             self.summary_writers[category] = tf.summary.FileWriter(filewriter_dir)
 
+    def _delete_all_events_files(self, directory_name: str) -> None:
```

Comment:
Minor: could add some logging to make the user aware this is happening.

Reply:
Added